letsencrypt/models/base_config_settings.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+from inspect import getmembers, isfunction
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    letsencrypt_dns_provider = fields.Selection(
+        selection='_get_supported_dns_providers')
+    letsencrypt_dns_login = fields.Char()
+    letsencrypt_dns_key = fields.Char()
+    letsencrypt_dns_url = fields.Char()
+    letsencrypt_needs_dns_provider = fields.Boolean()
+
+    def _get_supported_dns_providers(self):
+        providers = []
+        for func in [x for x in getmembers(
+            self, lambda x: isfunction(x) and
+                x.__name__.startswith('letsencrypt_') and
+                x.__name__.endswith('_support'))]:
+            provider = x[12:-8]
+            providers.append((provider, provider))
+        return providers
+
+    @api.multi
+    def default_get(self):
+        self.ensure_one()
```

Comment:
this fails, `default_get` is an @model method: https://github.com/OCA/OCB/blob/10.0/odoo/models.py#L1084

Reply:
you can't have tested this code at all, please actually run the config wizard an make it work